benchmarks/benchmarks/benchmark_utils.py

```diff
@@ -0,0 +1,15 @@
+from asv_runner.benchmarks.mark import SkipNotImplemented
+
+
+def skip_unsupported_datatype(queue, dtype):
+    """
+    Skip the benchmark if the device does not support the given data type.
+    """
+    if (
+        (dtype.name == "float64" or dtype.name == "complex128")
+        and not queue.sycl_device.has_aspect_fp64
+    ) or (dtype.name == "float16" and not queue.sycl_device.has_aspect_fp16):
+        raise SkipNotImplemented(
+            f"Skipping benchmark for {dtype.name} on this device"
+            + " as it is not supported."
+        )
```

Comment:
```suggestion
import dpctl.tensor as dpt
from asv_runner.benchmarks.mark import SkipNotImplemented


def skip_unsupported_dtype(q, dtype):
    """
    Skip the benchmark if the device does not support the given data type.
    """
    if (
        (dtype == dpt.float64 or dtype.name == dpt.complex128)
        and not q.sycl_device.has_aspect_fp64
    ) or (dtype == dpt.float16 and not q.sycl_device.has_aspect_fp16):
        raise SkipNotImplemented(
            f"Skipping benchmark for {dtype} on this device"
            + " as it is not supported."
        )
```

I suspect object comparison will be faster than string comparison, also changes make this more consistent with the rest of dpctl

Reply:
Implemented